pkg/domain/events.go

```diff
@@ -65,17 +65,29 @@ func (e EffectiveBlueprintCalculatedEvent) Message() string {
 }
 
 type StateDiffDeterminedEvent struct {
-	StateDiff StateDiff
+	Msg string
+}
+
+func newStateDiffComponentEvent(componentDiffs ComponentDiffs) StateDiffDeterminedEvent {
+	install, upgrade, uninstall, other := componentDiffs.Statistics()
+	msg := fmt.Sprintf("state diff determined: %d component diffs (%d to install, %d to upgrade, %d to delete, %d others)",
+		len(componentDiffs), install, upgrade, uninstall, other)
+	return StateDiffDeterminedEvent{Msg: msg}
+}
+
+func newStateDiffDoguEvent(doguDiffs DoguDiffs) StateDiffDeterminedEvent {
+	install, upgrade, uninstall, other := doguDiffs.Statistics()
+	msg := fmt.Sprintf("state diff determined: %d dogu diffs (%d to install, %d to upgrade, %d to delete, %d others)",
+		len(doguDiffs), install, upgrade, uninstall, other)
+	return StateDiffDeterminedEvent{Msg: msg}
 }
```

Comment:
Somehow using the same type for two things that are actually just a part of that type somehow feels semantically wrong.

Reply:
Hm... the events are inspired by domain events in DDD. In DDD others can react to events to use them as a hook for the next command to execute, so in clear DDD it would be better to have a single event for all diffs or a dedicated event type for each diff.

As we use kubernetes and kubernetes has no reliable event-API, we cannot react to events and they just stay at the CR to be better logs.

Reply:
> Hm... the events are inspired by domain events in DDD. In DDD others can react to events to use them as a hook for the next command to execute, so in clear DDD it would be better to have a single event for all diffs or a dedicated event type for each diff.
> 
> As we use kubernetes and kubernetes has no reliable event-API, we cannot react to events and they just stay at the CR to be better logs.

That sounds like a valid reason that justifies different event types, that other resources might wait for a dogu event but not a component event. I'll make a change accordingly :+1: